spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -236,6 +249,35 @@ class SpdxCompoundExpression(
             SpdxOperator.AND -> left.offersChoice() || right.offersChoice()
         }
 
+    override fun applyChoice(choice: SpdxExpression, subExpression: SpdxExpression): SpdxExpression {
+        if (!subExpression.validChoices().containsAll(choice.validChoices())) {
+            throw InvalidLicenseChoiceException(
+                "$choice is not a valid choice for the subtree expression. Valid choices are: ${validChoices()}."
```

Comment:
"subexpression" instead of "subtree expression", and I think it would be good to also print the subexpression in this message. Or maybe just say "...is not a valid choice for $subExpression. ..."